sources/light_client.move

```diff
@@ -523,3 +522,44 @@ public fun retarget_algorithm(p: &Params, previous_target: u256, first_timestamp
 
     next_target
 }
+
+
+/// Return the amount to `receiver_address` and the OP_RETURN message.
+/// If OP_RETURN is not included in the transaction, return an empty vector.
+/// * `height`: block heigh transacion belong
+/// * `proof`: merkle tree proof, this is the vector of 32bytes
+/// * `tx_index`: index of transaction in block
+/// * `transaction`: bitcoin transaction. Check transaction.move
+/// * `receiver_address`: address of receiver in p2pkh or p2wpkh
+/// @return address and amount for each output
+public fun prove_payment(
+    lc: &LightClient,
+    height: u64,
+    proof: vector<vector<u8>>,
+    tx_index: u64,
+    transaction: &Transaction,
+    receiver_address: vector<u8>
+) : (u256, vector<u8>, vector<u8>) {
+    let mut amount = 0;
+    let mut op_messages = vector[];
+    let tx_id = transaction.tx_id();
+    assert!(lc.verify_tx(height, tx_id, proof, tx_index), ETxNotInBlock);
+    let outputs = transaction.outputs();
+
+    let mut i = 0;
+
+    while (i < outputs.length()) {
+        let output = outputs[i];
+        if (output.is_pk_hash_script() && output.p2pkh_address() == receiver_address) {
+            amount = amount + output.amount();
+        };
+
+        if (output.is_op_return()) {
+            op_messages = output.op_return();
+        };
```

Comment:
```suggestion
        let o = outputs[i];
        if (o.is_pk_hash_script() && o.p2pkh_address() == receiver_address) {
            amount = amount + o.amount();
        };

        if (o.is_op_return()) {
            op_messages = o.op_return();
        };
```

Reply:
short names are good because it's less code, faster to read. However, short names only make sense if:
- the context is short (distance from declaration to usage is short)
- usage is clear
- declaration is clear eg, `let o = ouput[i]` is very clear.

Reply:
updated!